src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -150,7 +150,16 @@ public function generateUri($request, $path)
         // fortunately, they all are, so we have to remove entire query string
         $position = strpos($url, '?');
         if (false !== $position) {
+            $fragment = '';
```

Comment:
Could we use http://php.net/parse_url here instead of this custom logic?

Reply:
better thanks